### DISCUSSION TITLE
Keep car awake

### DISCUSSION THREAD

**Discussion by kjetilpp** [alandtse/tesla#212] [Q&A] [answered]
Hello. Is there a way to create a switch in HA, to force keep the car awake?

Example with battery below 20%, so can't use sentry mode, but need to keep car awake.

**Comment by alandtse** [comment, marked as answer]
https://github.com/alandtse/tesla/wiki/Polling-policy#polling-policy-always

**Comment by kjetilpp** [comment]
Thanks for the help! 
WAKE_UP did the trick.


Created this in node-red; 
![Screenshot 2022-05-04 211623](https://user-images.githubusercontent.com/5620578/166809496-2f1ee9f4-bcc6-4d53-8538-acdf68a87cf7.png)

`[
    {
        "id": "3b2a49591b0e3119",
        "type": "api-call-service",
        "z": "84c6cf4e64bcc6b1",
        "name": "Wake up stor bil",
        "server": "b0785337.16938",
        "version": 1,
        "debugenabled": true,
        "service_domain": "tesla_custom",
        "service": "api",
        "entityId": "",
        "data": "{\"command\":\"WAKE_UP\",\"parameters\":{\"path_vars\":{\"vehicle_id\":\"149302XXXXXXX00\"},\"wake_if_asleep\":true}}",
        "dataType": "json",
        "mergecontext": "",
        "output_location": "payload",
        "output_location_type": "none",
        "mustacheAltTags": false,
        "x": 400,
        "y": 220,
        "wires": [
            [
                "8d217579adb56090"
            ]
        ]
    },
    {
        "id": "024532f7bc992671",
        "type": "ha-entity",
        "z": "84c6cf4e64bcc6b1",
        "name": "Keep car awake",
        "server": "b0785337.16938",
        "version": 1,
        "debugenabled": false,
        "outputs": 2,
        "entityType": "switch",
        "config": [
            {
                "property": "name",
                "value": "Hold bil våken"
            },
            {
                "property": "device_class",
                "value": ""
            },
            {
                "property": "icon",
                "value": "mdi:car"
            },
            {
                "property": "unit_of_measurement",
                "value": ""
            }
        ],
        "state": "payload",
        "stateType": "msg",
        "attributes": [],
        "resend": true,
        "outputLocation": "payload",
        "outputLocationType": "none",
        "inputOverride": "allow",
        "x": 120,
        "y": 60,
        "wires": [
            [],
            []
        ]
    },
    {
        "id": "ed841e5889d3e526",
        "type": "server-state-changed",
        "z": "84c6cf4e64bcc6b1",
        "name": "",
        "server": "b0785337.16938",
        "version": 1,
        "exposeToHomeAssistant": false,
        "haConfig": [
            {
                "property": "name",
                "value": ""
            },
            {
                "property": "icon",
                "value": ""
            }
        ],
        "entityidfilter": "switch.hold_bil_vaken",
        "entityidfiltertype": "exact",
        "outputinitially": true,
        "state_type": "str",
        "haltifstate": "",
        "halt_if_type": "str",
        "halt_if_compare": "is",
        "outputs": 1,
        "output_only_on_state_change": true,
        "x": 210,
        "y": 140,
        "wires": [
            [
                "58867629408a3ddd"
            ]
        ]
    },
    {
        "id": "58867629408a3ddd",
        "type": "switch",
        "z": "84c6cf4e64bcc6b1",
        "name": "On Off",
        "property": "payload",
        "propertyType": "msg",
        "rules": [
            {
                "t": "eq",
                "v": "on",
                "vt": "str"
            },
            {
                "t": "eq",
                "v": "off",
                "vt": "str"
            }
        ],
        "checkall": "true",
        "repair": false,
        "outputs": 2,
        "x": 110,
        "y": 220,
        "wires": [
            [
                "3b2a49591b0e3119"
            ],
            [
                "92a98294259fe522"
            ]
        ]
    },
    {
        "id": "8d217579adb56090",
        "type": "delay",
        "z": "84c6cf4e64bcc6b1",
        "name": "",
        "pauseType": "delay",
        "timeout": "1",
        "timeoutUnits": "minutes",
        "rate": "1",
        "nbRateUnits": "1",
        "rateUnits": "second",
        "randomFirst": "1",
        "randomLast": "2",
        "randomUnits": "seconds",
        "drop": false,
        "allowrate": false,
        "outputs": 1,
        "x": 380,
        "y": 340,
        "wires": [
            [
                "6a5dbeefd6ae126a"
            ]
        ]
    },
    {
        "id": "6a5dbeefd6ae126a",
        "type": "api-current-state",
        "z": "84c6cf4e64bcc6b1",
        "name": "",
        "server": "b0785337.16938",
        "version": 1,
        "outputs": 2,
        "halt_if": "on",
        "halt_if_type": "str",
        "halt_if_compare": "is",
        "override_topic": false,
        "entity_id": "switch.hold_bil_vaken",
        "state_type": "str",
        "state_location": "payload",
        "override_payload": "msg",
        "entity_location": "data",
        "override_data": "msg",
        "blockInputOverrides": false,
        "x": 650,
        "y": 340,
        "wires": [
            [
                "3b2a49591b0e3119"
            ],
            []
        ]
    },
    {
        "id": "92a98294259fe522",
        "type": "change",
        "z": "84c6cf4e64bcc6b1",
        "name": "",
        "rules": [
            {
                "t": "set",
                "p": "reset",
                "pt": "msg",
                "to": "reset",
                "tot": "str"
            }
        ],
        "action": "",
        "property": "",
        "from": "",
        "to": "",
        "reg": false,
        "x": 140,
        "y": 380,
        "wires": [
            [
                "8d217579adb56090"
            ]
        ]
    },
    {
        "id": "b0785337.16938",
        "type": "server",
        "name": "Home Assistant",
        "legacy": false,
        "addon": true,
        "rejectUnauthorizedCerts": true,
        "ha_boolean": "y|yes|true|on|home|open",
        "connectionDelay": true,
        "cacheJson": true
    }
]`

**Comment by pandaedward** [comment]
@kjetilpp could you elaborate a bit more of making a `WAKE_UP` call please? I want to configure a similar button or switch specifically for waking up the tesla

**Comment by riesi77** [comment]
As far as I can see that, this is a direct api call from node-red. Would be great to have that possibility (wake the car up or set the car to sleep) also using this HA integration here.
